apis/python/src/tiledbsoma/sparse_nd_array.py

```diff
@@ -259,9 +271,7 @@ def read_table(self, coords: SparseNdCoordinates) -> Iterator[pa.Table]:
                     if lo_hi is not None:
                         sr.set_dim_ranges(dim_name, [lo_hi])
                 else:
-                    raise ValueError(
-                        f"could not handle coordinate with value {coord} of type {type(coord)}"
-                    )
+                    raise SOMAError(f"coord type {type(coord)} at slot {i} unhandled")
```

Comment:
This should be a TypeError, not a SOMAError.  In general, don't use SOMAError when there is a more corect built-in exception class.

See: https://docs.python.org/3/library/exceptions.html#TypeError

Relevant quote:
> Passing arguments of the wrong type (e.g. passing a [list](https://docs.python.org/3/library/stdtypes.html#list) when an [int](https://docs.python.org/3/library/functions.html#int) is expected) should result in a [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError), but passing arguments with the wrong value (e.g. a number outside expected boundaries) should result in a [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError).